contracts/protocol/bases/FundsBase.sol

```diff
@@ -265,16 +266,28 @@ abstract contract FundsBase is Context {
                     );
                 }
             } else {
+                uint256 exchangeId = _exchangeId; // stack too deep ToDO: any other way to avoid this?
+                uint256 nativeReturnAmount;
                 if (exchangeToken == address(0)) {
-                    IDRFeeMutualizer(exchange.mutualizerAddress).returnDRFee{ value: returnAmount }(
-                        _exchangeId,
-                        returnAmount
-                    );
+                    nativeReturnAmount = returnAmount;
                 } else {
                     if (returnAmount > 0) {
                         IERC20(exchangeToken).safeApprove(exchange.mutualizerAddress, returnAmount);
                     }
-                    IDRFeeMutualizer(exchange.mutualizerAddress).returnDRFee(_exchangeId, returnAmount);
+                }
+
+                try
+                    IDRFeeMutualizer(exchange.mutualizerAddress).returnDRFee{
+                        value: nativeReturnAmount,
+                        gas: RETURN_DR_FEE_GAS
+                    }(exchangeId, returnAmount)
+                {} catch {
```

Comment:
```suggestion
                {
                    emit IBosonFundsBaseEvents.DRFeeReturned(
                        _exchangeId,
                        exchangeToken,
                        returnAmount,
                        payable(exchange.mutualizerAddress),
                        sender
                    );
                } catch {
```
otherwise the `IBosonFundsBaseEvents.DRFeeReturned` event is never raised (problem was already there before the PR)